pkg/controller/che/che_controller.go

```diff
@@ -352,6 +352,15 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 	// Reconcile finalizers before CR is deleted
 	r.reconcileFinalizers(deployContext)
 
+	// Set CR Server Exposure strategy
+	if instance.Spec.Server.ServerExposureStrategy == "" {
+		strategy := util.GetServerExposureStrategy(instance)
+		instance.Status.OpenShiftOAuthUserCredentialsSecret = strategy
+		if err := r.UpdateCheCRSpec(instance, "serverExposureStrategy", strategy); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
```

Comment:
1. Move it to a new function `ensureServerExposureStrategy`.
2. Take into account that for k8s we have `spec.K8s.IngressStrategy`  field
so :
`instance.Spec.Server.ServerExposureStrategy == "" && instance.Spec.K8s.IngressStrategy == ""`
3. Add some tests for `ensureServerExposureStrategy`